cmd/csvlint/main.go

```diff
@@ -22,16 +24,17 @@ func main() {
 		printHelpAndExit(0)
 	}
 
-	var comma rune
-	switch *delimiter {
-	case "comma":
-		comma = ','
-	case "tab":
-		comma = '\t'
-	default:
-		fmt.Printf("unrecognized delimiter '%s'\n\n", *delimiter)
+	if flag.NFlag() > 0 {
+		fmt.Fprintln(os.Stderr, "Warning: not using defaults, may not validate CSV to RFC 4180")
+	}
+
+	converted_delimiter, err := strconv.Unquote(`'` + *delimiter + `'`)
```

Comment:
also, snake case should fail lint

but it looks like this file isn't linted in test